src/hci/gnomex/controller/DownloadSingleFileServlet.java

```diff
@@ -353,8 +353,7 @@ private FileDescriptor recurseGetMatchingFileDescriptor(FileDescriptor fd, Strin
     // Change all backslash to forward slash for comparison
     String fdFileName = fd.getFileName().replaceAll("\\\\", "/");
 
-
-    if (fdFileName.endsWith(fileName) && (dir.equals(theDirectory) || (dir.length() == 0 && theDirectory.equals("upload_staging")))) {
+    if (fdFileName.equals(fileName) || (dir.length() == 0 && theDirectory.equals("upload_staging"))) {

```

Comment:
In this method (recurseGetMatchingFileDescriptor), we are recursing through file directories even though file names should now be canonical. This method is only called in one place (in the area of lines 230-250 of the same file) which also appears to look through different directories. Perhaps this can be simplified now.